### ISSUE TITLE
Plan is not using correct value for current temperature

### ISSUE THREAD

**Issue by bwa** [mskg/homey-heating#144]
Using v1.2.4

**Describe the bug**
The plan is not using corrrect value as Current Temperature

**To Reproduce**
Create a Zone, create a Plan and add a temp sensor, 

**Expected behavior**
The Plan shall use the temperature from the only temperatur sensor in the zone

Note that it is not always wrong, but very often

The Termostate in the Zone is using correct temperature, the plan is not
**Screenshots**

[https://aws1.discourse-cdn.com/business4/uploads/athom/original/3X/6/c/6c7d32a6d3137ae0df8b091f16cbde40111823b8.png](url)

**Comment by mskg** [comment]
![image](https://user-images.githubusercontent.com/46903097/71594089-c6195a00-2b36-11ea-86f1-ccd1bcf1e9c2.png)

**Comment by mskg** [comment]
Plans do not read temperatures from any sensors, they only `set values` for devices supporting the `target_temperature` capability.

Virtual thermostats show the average of all devices, supporting the `current_temperature` capability being associated with a plan. 

If you now have two devices supporting `current_temperature` (Thermostat and Tempmaler), the value can different ((a+b) / 2) from a single reading. But that would not be an error.

I'm not quite sure if I understood the problem correctly.

**Comment by bwa** [comment]
But why are my Plans showing current temperature, and has the status of cooling or heating ?

![image](https://user-images.githubusercontent.com/170977/71601143-7ea0c700-2b52-11ea-9a22-adad1c3882cf.png)

**Comment by bwa** [comment]
To answer the other questions:

The only source of temperature reading in this zone, is the 'Sander Tempmåler', so noone trying to display the current_temperature cannot display other values that the Tempmåler is giving-

And the only device assosiated with the Heating plan is 
![image](https://user-images.githubusercontent.com/170977/71601808-61b9c300-2b55-11ea-98ca-f6b474d1d798.png)

**Comment by bwa** [comment]
Is it possible to debug or get some logs to see what's happening ? Meybe the only way to find the source of the strange reading ?

**Comment by mskg** [comment]
You also ticked the zone, such all devices in that zone are also automatically included. What kind of devices are you using?

**Comment by bwa** [comment]
Auara Temp sensors.

Ah, so I should not use both ?

BTW, fired up the app now on my computer. Got a lot of info :

********* APPLICATION STARTUP v0.0.0 (undefined) *********
[AthomAPI  ] Init
[APISvc    ] Connecting to API
Bootstrapping Driver v0.0.0 (undefined)
[AthomAPI  ] INIT global hooks
[AthomAPI  ] INIT device hooks
[AthomAPI  ] Found 5 devices
[AthomAPI  ] Listening on bddee17e-b06d-47a2-b2b9-c070e8033452 (Kjøkken Termostat) target_temperature
[AthomAPI  ] Listening on 9cc64268-edbb-4332-b195-6d1463803bab (VThermo Soverom) target_temperature
[AthomAPI  ] Listening on 9ce5aecf-8989-48d7-9e35-37422bfc75f9 (VThermo Bad) target_temperature
[AthomAPI  ] Listening on cb64bc43-9b06-4078-a663-291edaffc6ba (Sander Termostat) target_temperature
[AthomAPI  ] Listening on e7c78336-c60d-4ab9-85f7-94fe12ecdf32 (Silje Termostat) target_temperature
[AthomAPI  ] Listening on bddee17e-b06d-47a2-b2b9-c070e8033452 (Kjøkken Termostat) measure_temperature
[AthomAPI  ] Listening on 9cc64268-edbb-4332-b195-6d1463803bab (VThermo Soverom) measure_temperature
[AthomAPI  ] Listening on 9ce5aecf-8989-48d7-9e35-37422bfc75f9 (VThermo Bad) measure_temperature
[AthomAPI  ] Listening on cb64bc43-9b06-4078-a663-291edaffc6ba (Sander Termostat) measure_temperature
[AthomAPI  ] Listening on e7c78336-c60d-4ab9-85f7-94fe12ecdf32 (Silje Termostat) measure_temperature

**Comment by bwa** [comment]
Ups.... I startet npm run homey:run and when I stopped it :

✓ Uninstalling `app.mskg.homey-heating`...
✓ Homey App `app.mskg.homey-heating` successfully uninstalled

Did I kill my heating setup and app on my Homey ???

**Comment by mskg** [comment]
Yes, but that is normal behavior (for every App) if you just run it in development mode on your homey. Before you do that next time: Go to Settings in the App itself and backup your plans. Such you can always restore your setup. Devices will also not get lost that way, too.

There is also remote logging included in the App. You can enable it via the settings area, too. You get the same level of detail there.

**Comment by bwa** [comment]
Look like a new install from Homey App restored the setup.

So, is it possible to debug or trace at my computer without killing my install at Homey ?

**Comment by bwa** [comment]
Well, so first fix is to only use Device and not zone ?

**Comment by mskg** [comment]
> Well, so first fix is to only use Device and not zone ?

If there is only one device for your plan, just tick the device.

**Comment by bwa** [comment]
console.re ... that was neat ..

Is this something to bother : 

[AthomAPI ] Device c3f901ce-4da6-4d73-9ad9-1d69b4fd7257 was updated but cannot set temperature -> ignored

**Comment by mskg** [comment]
> [AthomAPI ] Device c3f901ce-4da6-4d73-9ad9-1d69b4fd7257 was updated but cannot set temperature -> ignored

No. It means that Homey updated device properties for that device but it has nothing to do with temperatures. We just ignore it.

**Comment by bwa** [comment]
Maybe the ZONES and DEVICES information in the Plan setup could be more clear about selecting both zones and devices. I have read through the text, and it is very far fetched that I should understand that selecting both would consider temp sensors twice.

**Comment by mskg** [comment]
> Maybe the ZONES and DEVICES information in the Plan setup could be more clear about selecting both zones and devices. I have read through the text, and it is very far fetched that I should understand that selecting both would consider temp sensors twice.

It doesn't consider something twice. You forgot that your thermostat also reports a temperature - that's the second one (in the same zone).

**Comment by bwa** [comment]
How can a thermostat report a 'current temperature' ? It's not a sensor, it's a kind of controller that need to read a sensor. This is very strange to understand.

**Comment by bwa** [comment]
BTW, I have created a flow that turns on and off the heaters wall-plug when thermostat is on or off. Is that correct way to 'connect' the termostate and the wall-plug ?

**Comment by mskg** [comment]
> How can a thermostat report a 'current temperature' ? It's not a sensor, it's a kind of controller that need to read a sensor. This is very strange to understand.

Because they have it, they report it.

**Comment by mskg** [comment]
> BTW, I have created a flow that turns on and off the heaters wall-plug when thermostat is on or off. Is that correct way to 'connect' the termostate and the wall-plug ?

Can you post some more details on your solution? Some screenshots of your flows?

**Comment by Allram** [comment]
I am experiencing the same problem here.
I have only ticket 'Zone', and some of my schedules/plans are showing the correct room temperature (actual temperature) those zones have virtual thermostats in them or Fibaro Radiator controller.

The zones that have Aqara Temp sensors are not showing the actual room temperature, but the plan seems to still be able to show heating/cooling dependin on the actual room temperature. Is it possible to add so that the actual room temperature is still showing in the plans device if an t.ex Aqara temp sensor is used? I guess they are not added as thermostats and thus not showing up correctly in the plans device?
